### PR TITLE
Add Dell XPS 17 9700

### PR DESCRIPTION
**Pull request by workflow**
Fixes https://github.com/NixOS/nixos-hardware/issues/290

See https://github.com/NixOS/nixos-hardware/pull/291

### CODE REVIEW

File: dell/xps/17-9700/common.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  # This will save you money and possibly your life!
+  services.thermald.enable = lib.mkDefault true;
+
+  boot.kernelPatches = [{
+    name = "enable-soundwire-drivers";
+    patch = null;
+    extraConfig = ''
+      SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES y
+      SND_SOC_INTEL_SOUNDWIRE_SOF_MACH m
+      SND_SOC_RT1308 m
```
Comment (Mic92) [inline dell/xps/17-9700/common.nix:19]: Can you also upstream these options to nixpkgs?

File: dell/xps/17-9700/common.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    '';
+    ignoreConfigErrors = true;
```
Comment (Mic92) [inline dell/xps/17-9700/common.nix:21]: What error are you seeing here?

File: dell/xps/17-9700/common.nix (hunk 1 above)
Comment (workflow) [reply dell/xps/17-9700/common.nix:19]: Good catch, seems to pass fine without now:
4a0cdc97c85cb941e8ff4a91226bbc92ca20cd9e

File: dell/xps/17-9700/common.nix (hunk 2 above)
Comment (workflow) [reply dell/xps/17-9700/common.nix]: Good catch, seems to pass fine without now:
4a0cdc97c85cb941e8ff4a91226bbc92ca20cd9e

File: dell/xps/17-9700/common.nix (hunk 1 above)
Comment (workflow) [reply dell/xps/17-9700/common.nix:19]: As default options for everyone?

Comment (Mic92) [reply dell/xps/17-9700/common.nix:19]: Yes. This is only building modules from the looks of it. So it's safe to have by default.

Comment (Mic92) [reply dell/xps/17-9700/common.nix:19]: Than users for this hardware no longer have to build their own kernel...

**Comment by IvanMalison** [comment]
Hey @workflow Sorry to ping you here, but wondering if you are running off of unstable/which graphics option you are using and if you have started encountering any diffculties as of late.

It's not clear to me what is going on, and interestingly my display manager seems to be working, but whenever i start an X11 graphical session it seems like things arent working anymore. Not sure what could have changed to make this happen.

**Comment by workflow** [comment]
@IvanMalison was still running on 22.05 when last I used the XPS, so probably wouldn't have run into it.
Maybe open a separate issue? Happy to test it at some point if you ping me there and hopefully help you reproduce 🤞

**Pull request merged**